### PR TITLE
Update to Xenial, expand matrix to cover new CMS releases

### PR DESCRIPTION
**Pull request by Cheddam**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,9 +30,10 @@ before_script:
 
   # Install composer dependencies
   - composer validate
-  - composer require --no-update silverstripe/recipe-core:1.0.x-dev
-  - if [[ $DB == PGSQL ]]; then composer require --no-update silverstripe/postgresql 2.0.x-dev; fi
-  - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+  - composer require --no-update silverstripe/recipe-core:$RECIPE_VERSION
+  - if [[ $DB == PGSQL ]]; then composer require --no-update silverstripe/postgresql:^2; fi
+  - if [[ $PHPCS_TEST ]]; then composer install --prefer-source --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile; fi
+  - if [[ ! $PHPCS_TEST ]]; then composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile; fi
```
Comment (dnsl48) [inline .travis.yml]: phpcs seems to be using framework rules

**Review by dnsl48 — CHANGES_REQUESTED** [review]
shall we patch phpcs call yet

File: .travis.yml (hunk 1 above)
Comment (Cheddam) [reply .travis.yml]: Yep, this is as good a time as any to shift them inline. Done.

**Review by dnsl48 — APPROVED** [review]
ace

**Pull request merged**